utils/query_builder.py

```diff
@@ -0,0 +1,96 @@
+from typing import List
+from utils.repository import convert_list_to_tuple_string
+
+
+class CosmosDBQueryBuilder:
+    query: str
+
+    def __init__(self):
+        super().__init__()
+        self.query = ""
+        self.parameters = []
+        self.select_conditions = []
+        self.where_conditions = []
+        self.limit = None
+        self.offset = None
+
+    def add_select_conditions(self, column: List[str] = None):
+        column = column if column else ["*"]
+        self.select_conditions.extend(column)
+        return self
+
+    def add_sql_in_condition(
+        self, attribute: str = None, ids_list: List[str] = None
+    ):
+        if ids_list and attribute and len(ids_list) > 0:
+            ids_values = convert_list_to_tuple_string(ids_list)
+            self.where_conditions.append(f"c.{attribute} IN {ids_values}")
+        return self
+
+    def add_sql_where_equal_condition(self, data: dict = None):
+        if data:
+            for k, v in data.items():
+                condition = f"c.{k} = @{k}"
+                self.where_conditions.append(condition)
+                self.parameters.append({'name': f'@{k}', 'value': v})
+        return self
+
+    def add_sql_visibility_condition(self, visible_only: bool):
+        if visible_only:
+            self.where_conditions.append('NOT IS_DEFINED(c.deleted)')
+        return self
+
+    def add_sql_limit_condition(self, limit):
+        if limit and isinstance(limit, int):
```

Comment:
Not sure if we need the isintance check.

Reply:
What if the person sends a string like "Hello"?
In that case, a syntax error would be generated when executing the query in the database

Reply:
In higher layers we can validate. Or later apply a stronger validation in the whole class. By now I think we are ok with the `if limit`  only. Same for the offset.